packages/frontend-api/src/Component/Constraints/CustomerUserRoleGroupAllowEditValidator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Component\Constraints;
+
+use Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
+use Shopsys\FrameworkBundle\Model\Customer\User\Role\CustomerUserRole;
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+use Symfony\Component\Validator\Exception\UnexpectedTypeException;
+
+/**
+ * @Annotation
+ */
+class CustomerUserRoleGroupAllowEditValidator extends ConstraintValidator
+{
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CurrentCustomerUser $currentCustomerUser
+     */
+    public function __construct(protected readonly CurrentCustomerUser $currentCustomerUser)
+    {
+    }
+
+    /**
+     * @param mixed $value
+     * @param \Symfony\Component\Validator\Constraint $constraint
+     */
+    public function validate($value, Constraint $constraint)
+    {
+        if (!$constraint instanceof CustomerUserRoleGroupAllowEdit) {
+            throw new UnexpectedTypeException($constraint, UniqueBillingAddressApi::class);
+        }
+
+        $currentCustomerUser = $this->currentCustomerUser->findCurrentCustomerUser();
+
+        if ($currentCustomerUser === null) {
+            return;
+        }
+
+        $customerUserGroupRoleUuid = $value;
+        $this->canEditCustomerUserRoleGroup($currentCustomerUser, $customerUserGroupRoleUuid, $constraint);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @param string $customerUserGroupRoleUuid
+     * @param \Shopsys\FrontendApiBundle\Component\Constraints\CustomerUserRoleGroupAllowEdit $constraint
+     */
+    protected function canEditCustomerUserRoleGroup(
+        CustomerUser $customerUser,
+        string $customerUserGroupRoleUuid,
+        CustomerUserRoleGroupAllowEdit $constraint,
+    ): void {
+        if (in_array(CustomerUserRole::ROLE_API_ALL, $customerUser->getRoles(), true)) {
```

Comment:
I found this warning about using `getRoles()` with hierarchical roles - https://symfony.com/doc/5.x/security.html#hierarchical-roles
I'm not sure if it applies to this case too, but I wanted to point it out

Reply:
Hmm, nice, it makes sense. Changed. 
At least, the check is simpler :angel:

Reply:
Hmm, nice, it makes sense. Changed. 
At least, the check is simpler :angel:

Reply:
* https://github.com/shopsys/shopsys/pull/3286/commits/58e89b0209a1aa7f7c7dffb22bd90fe8f1130c7a